src/behavior.ts

```diff
@@ -19,17 +22,18 @@ import { Stream } from "./stream";
 export type SemanticBehavior<A> = (time: Time) => A;
 
 @monad
-export abstract class Behavior<A> extends Reactive<A>
-  implements Observer<A>, Monad<A> {
+export abstract class Behavior<A> extends Reactive<A, BListener>
+  implements Parent<BListener>, Monad<A> {
   // Push behaviors cache their last value in `last`.
   // Pull behaviors do not use `last`.
   last: A;
-  nrOfListeners: number;
-  // The streams and behaviors that this behavior depends upon
-  dependencies: Cons<Reactive<any>>;
+  children: DoubleLinkedList<BListener> = new DoubleLinkedList();
   // Amount of nodes that wants to pull the behavior without actively
+  nrOfListeners: number;
   // listening for updates
   nrOfPullers: number;
+  pulledAt: number;
```

Comment:
`pulledAt` is also updated during push so the name is a bit misleading. Maybe we should call it `refreshedAt`, `updatedAt`, or something else?

Reply:
changedAt is updated when the internal `refresh` method is called, which would make `refreshedAt` a better name. `pulledAt` is actually describing last time someone checked if it has changed, so maybe `changedAt`?

Reply:
But, `changetAt` is not changed every time `refresh` is called. Only if the value changed.  I think `checkedAt` is a good name for `pulledAt`.